### PR TITLE
Introduce recorder auto_purge, remove purge_interval

### PR DESCRIPTION
**Pull request by amelchio**
## Type of change

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: home-assistant/core#33646
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

**Review by frenck — DISMISSED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

File: source/_integrations/recorder.markdown [hunk 1]
```diff
@@ -47,16 +47,16 @@ recorder:
       required: false
       default: 3
       type: integer
+    auto_purge:
+      description: Automatically purge the database every night. This keeps it from growing indefinitely. If you disable this, you can still use the service [`recorder.purge`](#service-purge) when required.
```
Comment (bdraco) [inline source/_integrations/recorder.markdown]:
```suggestion
      description: Automatically purge the database every night at 04:12 local time. Purging keeps the database from growing indefinitely, which will make Home Assistant slow. If you disable `auto_purge` it is recommended that you create an automation to call the [`recorder.purge`](#service-purge) periodically.
```

File: source/_integrations/recorder.markdown [hunk 2]
```diff
@@ -47,16 +47,16 @@ recorder:
       required: false
       default: 3
       type: integer
+    auto_purge:
+      description: Automatically purge the database every night at 04:12 local time. Purging keeps the database from growing indefinitely, which will make Home Assistant slow. If you disable `auto_purge` it is recommended that you create an automation to call the [`recorder.purge`](#service-purge) periodically.
```
Comment (amelchio) [inline source/_integrations/recorder.markdown]:
```suggestion
      description: Automatically purge the database every night at 04:12 local time. Purging keeps the database from growing indefinitely, which takes up disk space and can make Home Assistant slow. If you disable `auto_purge` it is recommended that you create an automation to call the [`recorder.purge`](#service-purge) periodically.
```

**Review by frenck — APPROVED** [review]
👍

**Pull request merged**